### ISSUE TITLE
Deep analysis of immutable types should convert immutables-generated contents of guava immutable collections

### ISSUE THREAD

**Issue by marquiswang** [immutables/immutables#333]
After #314, when ```@Value.Style(deepImmutablesDetection = true)``` is enabled , the constructor of @Value.Immutable implementation classes calls ```ImmutableFoo.copyOf()``` on fields that have @Value.Modifiable versions, but the elements of ```ImmutableList``` or other immutable collection fields do not get the same treatment.

Would it be possible, if we had a

```java
@Value.Style(deepImmutablesDetection = true)
@Value.Immutable
@Value.Modifiable
public interface Foo {
    List<Bar> getBars();

    // ...
}


@Value.Immutable
@Value.Modifiable
public interface Bar {
    // ...
}
```

For the ```ImmutableFoo.Builder#addBars``` and ```ImmutableFoo.Builder#addAllBars``` to call ```ImmutableBar.copyOf``` on the arguments as well?

**Comment by marquiswang** [comment]
Same issue with ```ModifiableFoo#from``` it should convert the contents of Immutable collection fields to Modifiable versions.

**Comment by elucash** [comment]
This is known limitation (as in javadoc `This have no effect on collection/container attributes to not interfere with invariant generic types.`). But we can remove the limitation, at least partially, we cannot have a covariant return `ImmutableList<ImmutableValue>` for `List<Value>` but we still can ensure that elements will be converted at runtime. The same for Modifiables.

**Comment by elucash** [comment]
Released in 2.3.7
It's hard for me to tell if it's enough here, I'll close this and let #462 track any remaining functionality
Thank you!
